catalog/app/containers/Bucket/PackageUpdateDialog.js

```diff
@@ -0,0 +1,933 @@
+            ),
+          })}
+        >
+          <input {...getInputProps()} />
+
+          {!!computedEntries.length && (
+            <div
+              className={cx(
+                classes.filesContainer,
+                !!error && classes.filesContainerErr,
+                !error && warn && classes.filesContainerWarn,
+              )}
+            >
+              {computedEntries.map(({ type, path, size }) => {
+                const action = {
+                  added: { hint: 'Remove', icon: 'clear', handler: rm },
+                  modified: { hint: 'Revert', icon: 'restore', handler: rm },
+                  deleted: { hint: 'Restore', icon: 'restore', handler: restore },
+                  unchanged: { hint: 'Delete', icon: 'clear', handler: del },
+                }[type]
+                return (
+                  // TODO: dif display for dif types
+                  // TODO: show delta for modified?
+                  <div
+                    key={`${type}:${path}`}
+                    className={cx(classes.fileEntry, classes[type])}
+                  >
+                    <M.IconButton
+                      onClick={handleAction(action.handler, path)}
+                      title={action.hint}
+                      size="small"
+                    >
+                      <M.Icon fontSize="inherit">{action.icon}</M.Icon>
+                    </M.IconButton>
+                    <div className={classes.filePath} title={path}>
+                      {path}
+                    </div>
+                    <div className={classes.fileSize}>{readableBytes(size)}</div>
+                  </div>
+                )
+              })}
+            </div>
+          )}
+
+          <div
+            className={cx(
+              classes.dropMsg,
+              !!error && classes.dropMsgErr,
+              !error && warn && classes.dropMsgWarn,
+            )}
+          >
+            {label}
+          </div>
+        </div>
+        {disabled && (
+          <div className={classes.lock}>
+            <div className={classes.progressContainer}>
+              <M.CircularProgress
+                size={80}
+                value={totalProgress.percent}
+                variant="determinate"
+              />
+              <div className={classes.progressPercent}>{totalProgress.percent}%</div>
+            </div>
+            <div className={classes.progressSize}>
+              {readableBytes(totalProgress.loaded)} / {readableBytes(totalProgress.total)}
+            </div>
+          </div>
+        )}
+      </div>
+    </div>
+  )
+}
+
+const getTotalProgress = R.pipe(
+  R.values,
+  R.reduce(
+    (acc, { progress: p = {} }) => ({
+      total: acc.total + (p.total || 0),
+      loaded: acc.loaded + (p.loaded || 0),
+    }),
+    { total: 0, loaded: 0 },
+  ),
+  (p) => ({
+    ...p,
+    percent: p.total ? Math.floor((p.loaded / p.total) * 100) : undefined,
+  }),
+)
+
+function DialogForm({
```

Comment:
Seems like it should be named `PackageUpdateDialog`

Reply:
`PackageUpdateDialog` is the module name, and in the context of the module, this component represents the form ui for this dialog (vs error, success and progress), not the whole dialog